pandas/tests/test_window.py

```diff
@@ -207,6 +207,26 @@ def f():
             'A', 'ra', 'std'), ('B', 'rb', 'mean'), ('B', 'rb', 'std')])
         tm.assert_frame_equal(result, expected, check_like=True)
 
+    def test_count_nonnumeric_types(self):
+        # GH12541
+        df_inf = DataFrame({'x': [1, 2, 3], 'y': [1., 2., np.Inf]})
+        df_date = DataFrame({'x': [1, 2, 3], 
+                            'y': pd.date_range('20130101',periods=3)})
```

Comment:
https://github.com/pandas-dev/pandas/pull/15054/files#diff-d1a05a7cf744e0feb0fc6cd7128903a8 is an example of how to create these various types (though need to add some nans)